stdlib/public/core/ClosedRange.swift

```diff
@@ -488,3 +488,18 @@ extension ClosedRange {
 // shorthand. TODO: Add documentation
 public typealias CountableClosedRange<Bound: Strideable> = ClosedRange<Bound>
   where Bound.Stride : SignedInteger
+
+extension ClosedRange: Codable where Bound: Codable {
```

Comment:
I think this should be implemented as two separate extensions:
```
extension ClosedRange: Encodable where Bound: Encodable {
...
}
extension ClosedRange: Decodable where Bound: Decodable {
...
}
```

Reply:
I disagree. Everywhere else a type conforms to both `Codable` and `Decodable` in the standard library or in the core the conformance is done in a single extension that conforms to `Codable`. 

You can see how the rest of the primitive types conform here: https://github.com/apple/swift/blob/2863b6cc642a44becb4b4586bf299e542101ef0c/stdlib/public/core/Codable.swift.gyb#L1639

Reply:
@nsalmoria I am actually curious though, what do you see as the benefit to splitting down the conformance into two separate extensions? Is there a particular reason behind that request?

Reply:
Unconditional conformance to Codable would be fine, but here the conformance is conditional. By requiring Bound to be Codable rather than just Encodable/Decodable, the condition is unnecessarily restrictive.

This is mentioned in the [CHANGELOG](https://github.com/apple/swift/blob/master/CHANGELOG.md):      

> Code that wants to be as precise as possible should generally not
>   conditionally conform to `Codable` directly, but rather its two constituent
>   protocols `Encodable` and `Decodable`, or else one can only (for instance)
>   decode a `Generic<Param>` if `Param` is `Encodable` in addition to
>   `Decodable`, even though `Encodable` is likely not required:
> 
>   ```swift
>   // Unnecessarily restrictive:
>   extension Generic: Codable where Param: Codable {}
> 
>   // More precise:
>   extension Generic: Encodable where Param: Encodable {}
>   extension Generic: Decodable where Param: Decodable {}
>   ```

Reply:
OK, that's a good point that I hadn't considered. I have made the appropriate changes: 
https://github.com/apple/swift/pull/19532/commits/15e8fd2b0fbe86cc2d4af59c4be927133b8a7208